frame/staking/src/lib.rs

```diff
@@ -746,17 +755,46 @@ enum Releases {
 	V5_0_0, // blockable validators.
 	V6_0_0, // removal of all storage associated with offchain phragmen.
 	V7_0_0, // keep track of number of nominators / validators in map
+	V8_0_0, // VoterList and efficient semi-sorted iteration
 }
 
 impl Default for Releases {
 	fn default() -> Self {
-		Releases::V7_0_0
+		Releases::V8_0_0
 	}
 }
 
 pub mod migrations {
 	use super::*;
 
+	pub mod v8 {
+		use super::*;
+
+		pub fn pre_migrate<T: Config>() -> Result<(), &'static str> {
+			ensure!(StorageVersion::<T>::get() == Releases::V7_0_0, "must upgrade linearly");
+			ensure!(VoterList::<T>::decode_len().unwrap_or_default() == 0, "voter list already exists");
+			Ok(())
+		}
+
+		pub fn migrate<T: Config>() -> Weight {
+			log!(info, "Migrating staking to Releases::V8_0_0");
+
+			let migrated = VoterList::<T>::regenerate();
+
+			StorageVersion::<T>::put(Releases::V8_0_0);
+			log!(
+				info,
+				"Completed staking migration to Releases::V8_0_0 with {} voters migrated",
+				migrated,
+			);
+
+			T::WeightInfo::regenerate(
+				CounterForValidators::<T>::get(),
+				CounterForNominators::<T>::get(),
+			).saturating_add(T::DbWeight::get().reads(2))
+		}
+	}
+
```

Comment:
@kianenigma what do you think of a quick PR to master move `mod migrations` to its own file ? This is getting big and is already its own module

Reply:
yeah, actually I'd be very happy if you make a `insubstantial` PR to master and split as much of staking as possible. Just having a single file sucks. I can think of: 


`pallet.rs` -> `mod pallet`
`pallet_impl` -> `impl<_> Pallet<_>`
`foo_impl` `impl<_> Foo<_> for Pallet<_>` (we have a bunch of these at least)
`migrations`